query-connector/src/app/query-service.ts

```diff
@@ -233,24 +238,54 @@ export async function parseFhirSearch(
   response: fetch.Response | Array<fetch.Response>,
   queryResponse: QueryResponse = {},
 ): Promise<QueryResponse> {
-  let resourceArray: any[] = [];
+  let resourceArray: unknown[] = [];
 
-  // Process the responses
+  // Process the responses and flatten them
   if (Array.isArray(response)) {
-    for (const r of response) {
-      resourceArray = resourceArray.concat(await processResponse(r));
-    }
+    resourceArray = (await Promise.all(response.map(processResponse))).flat();
   } else {
     resourceArray = await processResponse(response);
   }
 
   // Add resources to queryResponse
   for (const resource of resourceArray) {
-    const resourceType = resource.resourceType as keyof QueryResponse;
-    if (!queryResponse[resourceType]) {
-      queryResponse[resourceType] = [resource];
-    } else {
-      queryResponse[resourceType]!.push(resource);
+    if (isFhirResource(resource)) {
```

Comment:
question for my understanding: what's the reason we need to switch case this out compared to what we used to have?

Reply:
I don't think we necessarily need to; however, what I was running into was that the any here
``` let resourceArray: any[] = [];```
was getting flagged.
When you just create the the FhirResource, it introduces a new error because patient/observation/encounter can have different shapes.

The switch/case is definitely very very clunky, but I could not figure out another way to get rid of that error that also moved away from `any`.